src/Parameter.php

```diff
@@ -114,6 +114,8 @@ public function getAllowEmptyValue()
      * Default value is `false`.
      *
      * @param bool $allowEmptyValue
+     *
+     * @return Parameter
```

Comment:
What about adding php type hints instead?

Reply:
I thought about doing it. There are already places where they are used. But of course this will be without BC. So if it's OK with you I can add to all methods arguments and returns.

Reply:
As long as classes are final this is fine :)